File: PythonStmSrv/Model/Materie.py
class Materie:
    count = 0

    @classmethod
    def autoIncr(self):
        self.count += 1
        return self.count

    def __init__(self, denumire, profesor):
        self._mID = self.autoIncr()
        self._denumire = denumire
        self._profesor = profesor

    def getMID(self):
        return self._mID

    def setMID(self, mID):
        self._mID = mID

    def getDenumire(self):
        return self._denumire

    def setDenumire(self, denumire):
        self._denumire = denumire

    def getProfesor(self):
        return self._profesor

    def setProfesor(self, profesor):
        self._profesor = profesor

File: JavaJmsSrv/src/main/java/Model/Materie.java
package Model;

import java.util.concurrent.atomic.AtomicInteger;

public class Materie {

    private static final AtomicInteger count = new AtomicInteger(0);
    private int mID;
    private String denumire;
    private String profesor;


    public Materie(String denumire, String profesor) {
        this.mID = count.incrementAndGet();
        this.denumire = denumire;
        this.profesor = profesor;
    }

    public int getmID() {
        return mID;
    }

    public void setmID(int mID) {
        this.mID = mID;
    }

    public String getDenumire() {
        return denumire;
    }

    public void setDenumire(String denumire) {
        this.denumire = denumire;
    }

    public String getProfesor() {
        return profesor;
    }

    public void setProfesor(String profesor) {
        this.profesor = profesor;
    }
}
